java/vector/src/main/java/org/apache/arrow/vector/complex/DictionaryVector.java

```diff
@@ -21,24 +21,28 @@
 import io.netty.buffer.ArrowBuf;
 import org.apache.arrow.memory.BufferAllocator;
 import org.apache.arrow.memory.OutOfMemoryException;
+import org.apache.arrow.vector.BufferBacked;
+import org.apache.arrow.vector.FieldVector;
 import org.apache.arrow.vector.NullableIntVector;
 import org.apache.arrow.vector.ValueVector;
 import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.schema.ArrowFieldNode;
 import org.apache.arrow.vector.types.Dictionary;
 import org.apache.arrow.vector.types.Types.MinorType;
 import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.util.TransferPair;
 
 import java.util.HashMap;
 import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 
-public class DictionaryVector implements ValueVector {
+public class DictionaryVector implements FieldVector {
 
-  private ValueVector indices;
-  private Dictionary dictionary;
+  private final FieldVector indices;
```

Comment:
with the current implementation, the vector loaders/unloaders require a field vector. If the dictionary becomes a minor type we could probably change it back